corehq/apps/reports/views.py

```diff
@@ -1023,18 +1023,21 @@ def _get_case_url(case_id):
         tz_offset_ms = int(timezone.utcoffset(the_time_is_now).total_seconds()) * 1000
         tz_abbrev = timezone.localize(the_time_is_now).tzname()
 
-        # ledgers
+        product_name_by_id = {
+            product.product_id: product.name
+            for product in SQLProduct.objects.filter(domain=self.domain).values('product_id', 'name').all()
+        }
+
         def _product_name(product_id):
-            try:
-                return SQLProduct.objects.get(product_id=product_id).name
-            except SQLProduct.DoesNotExist:
-                return (_('Unknown Product ("{}")').format(product_id))
+            return product_name_by_id.get(product_id, _('Unknown Product ("{}")').format(product_id))
```

Comment:
why this functional changed.  i think `SQLProduct.objects.get(product_id=product_id).name` will also give a string name only right?

Reply:
There should be no functional change here. Before this change there was one query for every entry in the map. Now there is only one query to get all names and product ids. If there product_id does not exist, then we return "unknown product" if it does exist we return the name from `SQLProduct`

Reply:
COol.